src/createEnvfile.ts

```diff
@@ -46,10 +47,25 @@ export const createEnvFile = async (
     })
   );
 
+  let outputSecret: { [key: string]: string } | any = [];
```

Comment:
Thanks for PullRequest🐱

I also thought about as simple logic as possible🐱

1. We first define the following function.

```
const createSecretJsonsIncludedInWhitelist = (
  secretJsons: { [name: string]: any }[],
  outputWhitelist?: string[]
): { [name: string]: any }[] => {
  if (outputWhitelist === undefined) {
    return secretJsons;
  }

  return secretJsons.map((secretJson: { [name: string]: any }) => {
    let whitelistJson = {};
    for (const key of Object.keys(secretJson)) {
      if (outputWhitelist.includes(key)) {
        whitelistJson = Object.assign({ [key]: secretJson[key] }, whitelistJson);
      }
    }
    return whitelistJson;
  });
};
```

2. Next, modify `createEnvFile` as follows.

```
  const outputSecret = createSecretJsonsIncludedInWhitelist(secretJsons, params.outputWhitelist);

  const outputParams =
    params.addParams === undefined
      ? outputSecret
      : outputSecret.concat([params.addParams]);
```

3. Modify createDotEnv and createEnvrc as follows.

```
const createDotEnv = async (
  outputFile: string,
  outputParams: { [name: string]: any }[] | any,
  keyMapping?: { [name: string]: string }
): Promise<any> => {
  const appendFile = promisify(fs.appendFile);

  return await Promise.all(
    outputParams.map(async (outputParam: { [name: string]: any }) => {
      for (const [key, value] of Object.entries(outputParam)) {
        const keyName = keyMapping && keyMapping[key] ? keyMapping[key] : key;

        await appendFile(outputFile, `${keyName}=${value}\n`);
      }
    })
  );
};

const createEnvrc = async (
  outputFile: string,
  outputParams: { [name: string]: any }[] | any,
  keyMapping?: { [name: string]: string }
): Promise<any> => {
  const appendFile = promisify(fs.appendFile);

  return await Promise.all<any>(
    outputParams.map(async (outputParam: { [name: string]: any }) => {
      for (const [key, value] of Object.entries(outputParam)) {
        const keyName = keyMapping && keyMapping[key] ? keyMapping[key] : key;

        await appendFile(outputFile, `export ${keyName}=${value}\n`);
      }
    })
  );
};
```


I think that there are two good points of this method.

1. The original design has not changed significantly.
2. Complex array operations are hidden in `createSecretJsonsIncludedInWhitelist`.

Reply:
@keitakn 
Thank you for your review!
I fixed how to create whitelist.